File: Data Structures & Algorithms/binary-tree-diameter/submission-1.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func diameterOfBinaryTree(root *TreeNode) int {
    res := 0

    var dfs func(*TreeNode) int
    dfs = func (curr *TreeNode) int {
        if curr == nil {
            return 0
        }

        res = max(res, dfs(curr.Left) + dfs(curr.Right))
        return 1 + max(dfs(curr.Left), dfs(curr.Right))
    }

    dfs(root)
    return res
}

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        res = []
        def dfs(curr: TreeNode):
            if not curr:
                return

            nonlocal res
            res.append(curr.val)
            dfs(curr.left)
            dfs(curr.right)

        dfs(root)
        res = sorted(res)
        return res[k-1]
            